packages/types/src/websockets/collab.ts

```diff
@@ -0,0 +1,87 @@
+import z from 'zod';
+import type { Item } from '../items.js';
+import { WebSocketMessage } from './base.js';
+
+export const COLLAB = 'collab';
+export const COLLAB_COLORS = ['purple', 'pink', 'blue', 'green', 'yellow', 'orange', 'red'] as const;
+
+export type CollabColor = (typeof COLLAB_COLORS)[number];
+
+const BaseCollabMessage = WebSocketMessage.extend({
+	type: z.literal(COLLAB),
+	room: z.string(),
+});
+
+const ZodItem = z.custom<Partial<Item>>();
+
+// Incomming Messages
+export const WebSocketCollabMessage = z.discriminatedUnion('action', [
+	WebSocketMessage.extend({
+		type: z.literal(COLLAB),
+		action: z.literal('join'),
+		collection: z.string(),
+		item: z.string().nullable(),
+		version: z.string().nullable(),
+		initialChanges: ZodItem.optional(),
+	}),
+	BaseCollabMessage.extend({
+		action: z.literal('leave'),
+	}),
```

Comment:
Let's extract the `COLLAB` to one that contains multiple websocket message types, but leave out the changes for usages outside of collab for a separate branch.

Let's also extract the collab actions.